src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -57,8 +58,22 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
   }
 
   override def update(rows: Seq[Row]): IO[Unit] = {
-    val assetUpdates = rows.map(r => fromRow[AssetUpdate](r))
-    client.assets.update(assetUpdates) *> IO.unit
+    val assetUpdates = rows.map(r => fromRow[AssetsUpsertSchema](r))
+    val (updatesById, updatesByExternalId) = assetUpdates.partition(
+      p => p.id.isDefined
+    )
+    val id = if (updatesById.isEmpty) { IO.unit } else {
+      client.assets.updateById(updatesById.map(a => a.id.get -> a.transformInto[AssetUpdate]).toMap)
+    }
+    val extId = if (updatesByExternalId.isEmpty) { IO.unit } else {
```

Comment:
`val updateExternalds` because this is an effect that does something when you run it.